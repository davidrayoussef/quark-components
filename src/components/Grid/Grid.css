.grid {
  display: grid;
  grid-gap: 10px;
  grid-template-columns: var(--cols);
  grid-template-rows: var(--rows);
}

.grid.fluid {
  grid-template-columns: repeat(auto-fit, minmax(150px, 1fr));
  grid-auto-rows: 50px;
}
