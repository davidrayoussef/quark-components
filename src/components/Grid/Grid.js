import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import style from './Grid.css';

const Grid = ({ children, cols, rows, fluid }) => {
  cols = typeof cols === 'string' ? cols : typeof cols === 'number' ? `repeat(${cols}, 1fr)` : null;
  rows = typeof rows === 'string' ? rows : typeof rows === 'number' ? `repeat(${rows}, 1fr)` : null;

  return (
    <Fragment>
      <div className={`${style.grid} ${fluid ? style.fluid : ''}`}>
        {children}
      </div>
      <style>
        {`
          :root {
            ${cols ? '--cols: ' + cols : ''};
            ${rows ? '--rows: ' + rows : ''};
          }
        `}
      </style>
    </Fragment>
  );
};

Grid.propTypes = {
  children: PropTypes.node.isRequired,
  cols: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.string
  ]),
  rows: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.string
  ]),
  fluid: PropTypes.bool
};

export default Grid;
