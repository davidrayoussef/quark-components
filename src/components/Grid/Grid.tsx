import React, { Fragment } from 'react';

import { GridProps } from '../../shared';

import style from './Grid.scss';

export const Grid: React.FC<GridProps> = ({
  children,
  cols,
  rows,
  fluid,
  ...rest
}: GridProps) => {
  cols =
    typeof cols === 'string'
      ? cols
      : typeof cols === 'number'
      ? `repeat(${cols}, 1fr)`
      : undefined;
  rows =
    typeof rows === 'string'
      ? rows
      : typeof rows === 'number'
      ? `repeat(${rows}, 1fr)`
      : undefined;
  return (
    <Fragment>
      <div className={`${style.grid} ${fluid ? style.fluid : ''}`} {...rest}>
        {children}
      </div>
      <style>
        {`
          :root {
            ${cols ? '--cols: ' + cols : ''};
            ${rows ? '--rows: ' + rows : ''};
          }
        `}
      </style>
    </Fragment>
  );
};
