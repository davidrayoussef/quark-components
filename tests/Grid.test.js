import React from 'react';
import { shallow } from 'enzyme';
import Grid from '../src/components/Grid';

test('should render', () => {
  const wrapper = shallow(
    <Grid>
      <div></div>
      <div></div>
      <div></div>
    </Grid>
  );
  expect(wrapper);
});

test('should add class .fluid if fluid prop set', () => {
  const wrapper = shallow(
    <Grid fluid>
      <div></div>
      <div></div>
      <div></div>
    </Grid>
  );
  expect(wrapper.find('.fluid').length).toBe(1);
});

test('should render correct amount of children', () => {
  const wrapper = shallow(
    <Grid>
      <li></li>
      <li></li>
      <li></li>
      <li></li>
      <li></li>
      <li></li>
    </Grid>
  );
  expect(wrapper.find('li').length).toBe(6);
});

test('snapshot', () => {
  const wrapper = shallow(
    <Grid>
      <div></div>
      <div></div>
      <div></div>
    </Grid>
  );
  expect(wrapper).toMatchSnapshot();
});
