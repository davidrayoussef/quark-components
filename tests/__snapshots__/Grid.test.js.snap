// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<React.Fragment>
  <div
    className="grid "
  >
    <div />
    <div />
    <div />
  </div>
  <style>
    
          :root {
            ;
            ;
          }
        
  </style>
</React.Fragment>
`;
